web/src/app/pages/create-survey/task-details/task-input/task-input.component.ts

```diff
@@ -0,0 +1,354 @@
+import { TaskGroup } from '../task-details.component';
+
+
+export interface TaskTypeSelectOption {
+  icon: string;
+  label: string;
+  type: TaskType;
+  cardinality?: Cardinality;
+}
+
+@Component({
+  selector: 'task-input',
+  templateUrl: './task-input.component.html',
+  styleUrls: ['./task-input.component.scss'],
+})
+export class TaskInputComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() label?: string;
+  @Input() required?: boolean;
+  @Input() taskType?: TaskType;
+  @Input() multipleChoice?: MultipleChoice;
+  @Input() cardinality?: Cardinality;
+  @Input() taskCount?: Number;
+  @Output() update = new EventEmitter();
+  @Output() delete = new EventEmitter();
+  taskOptions: MultipleChoice | undefined;
+  selectTaskOptions: TaskTypeSelectOption[];
+  @Input() taskIndex?: number;
+  // TODO: This name TaskGroup is ambigous. We need to resolve this in
+  // relation to the TaskType above.
+  @Input() group?: TaskGroup;
+
+  taskGroup: FormGroup;
+  @ViewChild('questionInput', { static: true }) questionInput?: ElementRef;
+
+  /** When expanded, options and actions below the fold are visible to the user. */
+  expanded: boolean;
+
+  /** Set to true when question gets focus, false when it loses focus. */
+  selected: boolean;
+
+  subscription: Subscription = new Subscription();
+
+  @HostListener('click')
+  onTaskFocus() {
+    this.expanded = true;
+    this.selected = true;
+  }
+
+  @HostListener('document:click')
+  onTaskBlur() {
+    if (!this.selected) {
+      this.expanded = false;
+    }
+    this.selected = false;
+  }
+
+  constructor(
+    private taskBuilder: FormBuilder,
+    private dialogService: DialogService,
+    private jobService: JobService,
+    private readonly cdr: ChangeDetectorRef
+  ) {
+    this.expanded = false;
+    this.selected = false;
+    this.taskType = TaskType.TEXT;
+    this.selectTaskOptions = [
+      {
+        icon: 'short_text',
+        label: 'Text',
+        type: TaskType.TEXT,
+      },
+      {
+        icon: 'radio_button_checked',
+        label: 'Select One',
+        type: TaskType.MULTIPLE_CHOICE,
+        cardinality: Cardinality.SELECT_ONE,
+      },
+      {
+        icon: 'library_add_check',
+        label: 'Select multiple',
+        type: TaskType.MULTIPLE_CHOICE,
+        cardinality: Cardinality.SELECT_MULTIPLE,
+      },
+      {
+        icon: 'photo',
+        label: 'Photo',
+        type: TaskType.PHOTO,
+      },
+      {
+        icon: 'tag',
+        label: 'Number',
+        type: TaskType.NUMBER,
+      },
+      {
+        icon: 'calendar_today',
+        label: 'Date',
+        type: TaskType.DATE,
+      },
+      {
+        icon: 'access_time',
+        label: 'Time',
+        type: TaskType.TIME,
+      },
+    ];
+    this.taskGroup = this.taskBuilder.group({
+      label: ['', this.validateLabel.bind(this)],
+      required: [false],
+      // By default we set the select task to be of text type.
```

Comment:
If tasks are always added via one of the add task buttons, what does it mean for a "default task" to be added? Is this actually the default for "question" tasks?

Reply:
Default Task in this case is TaskType.TEXT, which is just so that there is a pre-selected choice in the drop down.

Then, the user can change it to TaskType.MULTIPLE_CHOICE, TaskType.NUMBER and so on.

Reply:
I think I might be confused because the task type in the UI is different from task type in the code. From the user's perspective, this is the "default question type". Not sure how to resolve this so LGTM for now.

Reply:
Actually I think something is off here. "TEXT" is the default for "QUESTION" tasks, but there can be other task types with other subtypes (suggest loi can be of subtype DROP PIN, CAPTURE LOCATION, DRAW POLYGON). I'll LGTM in case you want to address this is a separate PR.

Reply:
Right, I think this is just something wrong with the general integration fo the old code into the new model, where Question is not the only thing? Idk. But happy to defer this to another PR.